SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -593,7 +593,7 @@ void ASpatialDebugger::EditorInitialiseWorkerRegions()
 	const ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(World->GetWorldSettings());
 	check(WorldSettings != nullptr);
 
-	const TSubclassOf<UAbstractSpatialMultiWorkerSettings> MultiWorkerSettingsClass = WorldSettings->MultiWorkerSettingsClass;
+	const TSubclassOf<UAbstractSpatialMultiWorkerSettings> MultiWorkerSettingsClass = WorldSettings->GetMultiWorkerSettingsClass();
```

Comment:
I think there's actually a bug here - this code should respect the enable multiworker flag - similar to what's done in `CreateAndInitializeLoadBalancingClasses`. Could we centralise this logic?

Reply:
The check if the multiworker flag is enabled is currently wrapped around this function, so I don't think there is a bug. But if the current implementation is not clear I can refactor it?
```
void ASpatialDebugger::EditorRefreshWorkerRegions()
{
	DestroyWorkerRegions();

	if (bShowWorkerRegions && EditorAllowWorkerBoundaries())
	{
		EditorInitialiseWorkerRegions();
		CreateWorkerRegions();
	}

	EditorRefreshDisplay();
}

bool ASpatialDebugger::EditorAllowWorkerBoundaries() const
{
	// Check if multi worker is enabled.
	UWorld* World = GetWorld();
	check(World != nullptr);

	const bool bIsMultiWorkerEnabled = USpatialStatics::IsSpatialMultiWorkerEnabled(World);
	const bool bIsSpatialNetworkingEnabled = GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking();

	return bIsMultiWorkerEnabled && bIsSpatialNetworkingEnabled;
}
```

Reply:
We are missing this behaviour from `CreateAndInitializeLoadBalancingClasses ` in `EditorInitialiseWorkerRegions()` to show one worker if no load balancing strategy is set. We can add this but probably fits better with PR2505.
```
	// If multi worker is disabled, the USpatialMultiWorkerSettings CDO will give us single worker behaviour.
	const TSubclassOf<UAbstractSpatialMultiWorkerSettings> MultiWorkerSettingsClass =
		bMultiWorkerEnabled ? *WorldSettings->GetMultiWorkerSettingsClass() : USpatialMultiWorkerSettings::StaticClass();
```

Reply:
I've centralised all the logic related to the `MultiWorkerSettingsClass ` into the SpatialWorldSettings and provided accessor functions in SpatialStatics to avoid duplicate code.